x-pack/plugins/alerts/server/saved_objects/migrations.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import {
+  SavedObjectMigrationMap,
+  SavedObjectUnsanitizedDoc,
+  SavedObjectMigrationFn,
+  Logger
+} from '../../../../../src/core/server';
+import { RawAlert } from '../types';
+import { EncryptedSavedObjectsPluginSetup } from '../../../encrypted_saved_objects/server';
+
+export function getMigrations(
+  encryptedSavedObjects: EncryptedSavedObjectsPluginSetup,
+  eventLogger: Logger
+): SavedObjectMigrationMap {
+  const migrations: SavedObjectMigrationMap = {};
+
+  const alertsMigration = changeAlertingConsumer(encryptedSavedObjects, 'alerting', 'alerts', eventLogger);
+  if (alertsMigration) {
+    migrations['7.9.0'] = alertsMigration;
+  }
+
+  const infrastructureMigration = changeAlertingConsumer(encryptedSavedObjects, 'metrics', 'infrastructure', eventLogger);
+  if (infrastructureMigration) {
+    migrations['7.10.0'] = infrastructureMigration;
+  }
```

Comment:
I think there should always be a migration returned but we may have to wrap it within another function that does the try/catch. It would also provide better information for error messages.

For example:
```
const infrastructureMigration = changeAlertingConsumer(encryptedSavedObjects, 'metrics', 'infrastructure', eventLogger);
migrations['7.10.0'] = (doc) => {
    try {
        return infrastructureMigration(doc);
    } catch (e) {
        eventLogger.error(`encryptedSavedObject migration failed for alert ${doc.id} with error: ${ex.message}`);
    }
}
```